### PR TITLE
fix: do not hard-code chrome for interactive e2e

### LINKED ISSUE #15808: open-ct complains about missing browser but open works fine
### Current behavior

`npx cypress open` works just fine but `npx cypress open-ct` crashes with:

```
(node:33115) Warning: Accessing non-existent property 'padLevels' of module exports inside circular dependency
(Use `Cypress --trace-warnings ...` to show where the warning was created)
Error: Can't run because you've entered an invalid browser name.

Browser: 'chrome' was not found on your system or is not supported by Cypress.

Cypress supports the following browsers:
- chrome
- chromium
- edge
- electron
- firefox

You can also use a custom browser: https://on.cypress.io/customize-browsers

Available browsers found on your system are:
- chromium
- firefox
- electron
    at get (~/.cache/Cypress/7.0.0/Cypress/resources/app/packages/server/lib/errors.js:995:15)
    at Object.throwErr [as throw] (~/.cache/Cypress/7.0.0/Cypress/resources/app/packages/server/lib/errors.js:1013:9)
    at throwBrowserNotFound (~/.cache/Cypress/7.0.0/Cypress/resources/app/packages/server/lib/browsers/index.js:140:22)
    at ~/.cache/Cypress/7.0.0/Cypress/resources/app/packages/server/lib/browsers/index.js:123:12
    at tryCatcher (~/.cache/Cypress/7.0.0/Cypress/resources/app/packages/launcher/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (~/.cache/Cypress/7.0.0/Cypress/resources/app/packages/launcher/node_modules/bluebird/js/release/promise.js:512:31)
    at Promise._settlePromise (~/.cache/Cypress/7.0.0/Cypress/resources/app/packages/launcher/node_modules/bluebird/js/release/promise.js:569:18)
    at Promise._settlePromise0 (~/.cache/Cypress/7.0.0/Cypress/resources/app/packages/launcher/node_modules/bluebird/js/release/promise.js:614:10)
    at Promise._settlePromises (~/.cache/Cypress/7.0.0/Cypress/resources/app/packages/launcher/node_modules/bluebird/js/release/promise.js:694:18)
    at _drainQueueStep (~/.cache/Cypress/7.0.0/Cypress/resources/app/packages/launcher/node_modules/bluebird/js/release/async.js:138:12)
    at _drainQueue (~/.cache/Cypress/7.0.0/Cypress/resources/app/packages/launcher/node_modules/bluebird/js/release/async.js:131:9)
    at Async._drainQueues (~/.cache/Cypress/7.0.0/Cypress/resources/app/packages/launcher/node_modules/bluebird/js/release/async.js:147:5)
    at Immediate.Async.drainQueues [as _onImmediate] (~/.cache/Cypress/7.0.0/Cypress/resources/app/packages/launcher/node_modules/bluebird/js/release/async.js:17:14)
    at processImmediate (internal/timers.js:461:21)
 {
  isCypressErr: true,
  type: 'BROWSER_NOT_FOUND_BY_NAME',
  details: undefined
}
Error: Can't run because you've entered an invalid browser name.

Browser: 'chrome' was not found on your system or is not supported by Cypress.

Cypress supports the following browsers:
- chrome
- chromium
- edge
- electron
- firefox

You can also use a custom browser: https://on.cypress.io/customize-browsers

Available browsers found on your system are:
- chromium
- firefox
- electron
    at get (~/.cache/Cypress/7.0.0/Cypress/resources/app/packages/server/lib/errors.js:995:15)
    at Object.throwErr [as throw] (~/.cache/Cypress/7.0.0/Cypress/resources/app/packages/server/lib/errors.js:1013:9)
    at throwBrowserNotFound (~/.cache/Cypress/7.0.0/Cypress/resources/app/packages/server/lib/browsers/index.js:140:22)
    at ~/.cache/Cypress/7.0.0/Cypress/resources/app/packages/server/lib/browsers/index.js:123:12
    at tryCatcher (~/.cache/Cypress/7.0.0/Cypress/resources/app/packages/launcher/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (~/.cache/Cypress/7.0.0/Cypress/resources/app/packages/launcher/node_modules/bluebird/js/release/promise.js:512:31)
    at Promise._settlePromise (~/.cache/Cypress/7.0.0/Cypress/resources/app/packages/launcher/node_modules/bluebird/js/release/promise.js:569:18)
    at Promise._settlePromise0 (~/.cache/Cypress/7.0.0/Cypress/resources/app/packages/launcher/node_modules/bluebird/js/release/promise.js:614:10)
    at Promise._settlePromises (~/.cache/Cypress/7.0.0/Cypress/resources/app/packages/launcher/node_modules/bluebird/js/release/promise.js:694:18)
    at _drainQueueStep (~/.cache/Cypress/7.0.0/Cypress/resources/app/packages/launcher/node_modules/bluebird/js/release/async.js:138:12)
    at _drainQueue (~/.cache/Cypress/7.0.0/Cypress/resources/app/packages/launcher/node_modules/bluebird/js/release/async.js:131:9)
    at Async._drainQueues (~/.cache/Cypress/7.0.0/Cypress/resources/app/packages/launcher/node_modules/bluebird/js/release/async.js:147:5)
    at Immediate.Async.drainQueues [as _onImmediate] (~/.cache/Cypress/7.0.0/Cypress/resources/app/packages/launcher/node_modules/bluebird/js/release/async.js:17:14)
    at processImmediate (internal/timers.js:461:21)
```

### Versions

- Cypress 7.0.0
- Fedora GNU/Linux

### PR DESCRIPTION
**Pull request by lmiller1990**
- Closes #15808

### User facing changelog

Do not assume chrome exists on every machine when launching the component testing runner in interactive mode.

### Additional details

Some machines will not have chrome installed. In that case, we fall back to chromium -> edge -> electron etc.

### How has the user experience changed?

Do not throw an error when running `yarn cypress open-ct` if Chrome is not installed.

### PR Tasks

- [x] Have tests been added/updated?
- [x] Has the original issue or this PR been tagged with a release in ZenHub?  **7.0.1**

### CODE REVIEW

File: packages/server/lib/modes/interactive-ct.js [hunk 1]
```diff
@@ -1,13 +1,32 @@
...
+const browsersForCtInteractive = ['chrome', 'chromium', 'edge', 'electron', 'firefox']
 
-  options.browser = options.browser || 'chrome'
+const returnDefaultBrowser = (browsersByPriority, installedBrowsers) => {
+  const browserMap = installedBrowsers.reduce((acc, curr) => {
+    acc[curr.name] = true
 
-  return serverCt.start(projectRoot, options)
+    return acc
+  }, {})
```
Comment (agg23) [inline packages/server/lib/modes/interactive-ct.js:11]: This should be a `Set` instead

**Review by agg23 — APPROVED** [review]
Looks good other than the nit about `Set`

File: packages/server/lib/modes/interactive-ct.js (hunk 1 above)
Comment (lmiller1990) [reply packages/server/lib/modes/interactive-ct.js:11]: Is there an objective reason that is better?

Comment (agg23) [reply packages/server/lib/modes/interactive-ct.js:11]: It's cleaner and more clear

Comment (lmiller1990) [reply packages/server/lib/modes/interactive-ct.js:11]: Oops I merged, I'll make this change when I'm in this area again (prob soon). Thanks!

**Pull request merged**